### PR TITLE
chore(weave): Rag llm judge tutorial

### PR DESCRIPTION
**Pull request by scottire**

### CODE REVIEW

File: docs/docs/tutorial-rag.md [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+When there aren't simple ways to evaluate your application, one approach is to use an LLM to evaluate aspects of it. Here is an example of using an LLM judge to try to measure the context precision by prompting it to verify if the context was useful in arriving at the given answer. This prompt was augmented from the popular [RAGAS framework](https://docs.ragas.io/). 
+
+### Defining a scoring function
+
+As we did in the [Build an Evaluation pipeline tutorial](/tutorial-eval), we'll define a set of example rows to test our app against and a scoring function. Our scoring function will take one row and evaluate it. The input arguments should match with the corresponding keys in our row, so `question` here will be taken from the row dictionary. `prediction` is the output of the model. The input to the model will be taken from the example based on its input argument, so `question` here too. 
+
+```python
+# highlight-next-line
+@weave.op()
+async def context_precision_score(question, prediction):
```
Comment (NiWaRe) [inline docs/docs/tutorial-rag.md:116]: @scottire this should change to `model_output` from v49 I think. Maybe also a quick overview of the data flow in the eval pipeline (from dataset columns, to model pred, and eval scorers) as overview would help?

File: docs/docs/tutorial-rag.md [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+        answer = response.choices[0].message.content
+        return {'answer': answer, 'context': context}
+
+weave.init('rag-qa')
+model = RAGModel(
+    system_message="You are an expert in finance and answer questions related to finance, financial services, and financial markets. When responding based on provided information, be sure to cite the source."
+)
+
+@weave.op()
+async def context_precision_score(question, prediction):
```
Comment (NiWaRe) [inline docs/docs/tutorial-rag.md:242]: @scottire prediction -> model_output

File: docs/docs/tutorial-rag.md [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+    {"question": "How much did Berkshire Hathaway's cash levels increase in the fourth quarter?"},
+    {"question": "What is the goal of Highmark Health's integration of Google Cloud and Epic Systems technology?"},
+    {"question": "What were Rivian and Lucid's vehicle production forecasts for 2024?"},
+    {"question": "Why was the Norwegian Dawn cruise ship denied access to Mauritius?"},
+    {"question": "Which company achieved the first U.S. moon landing since 1972?"},
+    {"question": "What issue did Intuitive Machines' lunar lander encounter upon landing on the moon?"}
+]
+
+evaluation = weave.Evaluation(dataset=questions, scorers=[context_precision_score])
+import asyncio
```
Comment (NiWaRe) [inline docs/docs/tutorial-rag.md:280]: @scottire quick explanation or link to docs would be good here, at least for me before weave I never really used async before

File: docs/docs/tutorial-rag.md [hunk 4]
```diff
@@ -0,0 +1,287 @@
+# Example: Model Based Evaluation on RAG applications
+
+Retrieval Augmented Generation (RAG) is a common approach of building Generative AI applications that have access to custom knowledge bases. 
+
+In this example, we'll show an example that has a retrieval step to get documents. By tracking this, you can debug your app and see what documents were pulled into the LLM context.
+We'll also show how to evaluate it using an LLM judge.
+
+## Build a knowledge base
```
Comment (NiWaRe) [inline docs/docs/tutorial-rag.md:8]: @scottire the sections are very nice and clear - adding some numbers to them 1, 2a, 2b, etc. with a first index/summary would help me to put the steps together

File: docs/docs/tutorial-rag.md [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+    {"question": "Which company achieved the first U.S. moon landing since 1972?"},
+    {"question": "What issue did Intuitive Machines' lunar lander encounter upon landing on the moon?"}
+]
+
+evaluation = weave.Evaluation(dataset=questions, scorers=[context_precision_score])
+import asyncio 
+asyncio.run(evaluation.evaluate(model))
+```
+
+# Conclusion
```
Comment (NiWaRe) [inline docs/docs/tutorial-rag.md:284]: @scottire here as "What's next" we could point them to a full repo to clone and start with -- especially for beginners having this central config file to just pop in HF models and use pre-defined metrics could be interesting. I could change mine to not have LC for example (still working on making it simpler and easier)?

File: docs/docs/tutorial-rag.md [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+        )
+        document_embeddings.append(response)
+    return document_embeddings
+
+article_embeddings = docs_to_embeddings(articles) # Note: you would typically do this once with your articles and put the embeddings & metadata in a database
+```
+
+## Create a RAG app
+
+Next, we wrap our retrieval function `get_most_relevant_document` with a `weave.op()` decorator and we create our `Model` class. We call `weave.init('rag-qa')` to begin tracking all the inputs and outputs of our functions for later inspection.
```
Comment (jamie-rasmussen) [inline docs/docs/tutorial-rag.md:47]: Tim was just changing all of the project names across the docs to `intro-example` - could we do that here?

File: docs/docs/tutorial-rag.md [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+# Pulling it all together
+
+To get the same result for your RAG apps:
+- Wrap LLM calls & retrieval step functions with `weave.op()`
+- (optional) Create a `Model` subclass with `predict` function and app details
+- Collect examples to evaluate
+- Create scoring functions that score one example
+- Use `Evaluation` class to run evaluations on your examples
+
+Here, we show the code in it's entirety.
```
Comment (jamie-rasmussen) [inline docs/docs/tutorial-rag.md:177]: it's -> its

**Review by jamie-rasmussen — APPROVED** [review]

**Pull request merged**